components/Main/index.js

```diff
@@ -1,22 +1,155 @@
-import { getLocalStorage } from "../../helpers/localStorage.js";
-import { getProjectFromLocalStorageById } from "../../services/project.js";
+import {
+  getLocalStorage,
+  setLocalStorage,
+} from "../../helpers/localStorage.js";
+import {
+  getProjectById,
+  deleteProject,
+  updateProjects,
+} from "../../services/project.js";
+import { previewProjectsNameOnTheSidebar } from "../Sidebar/index.js";
+import { renderCurrentProject } from "../Sidebar/index.js";
 let proName = document.querySelector(".project-name");
+//  AA!243232
+const threeDotsIcon = document.querySelector(".three-dots-icon");
+const redDeleteIcon = document.querySelector(".red-delete-icon");
+const greenTrueIcon = document.querySelector(".green-true-icon");
+let wantedProject;
 export const viewProjectNameOnMainPage = () => {
-  let wantedProject = getProjectFromLocalStorageById(
-    +localStorage.getItem("currentProjectIdFromSidebar")
+  wantedProject = getProjectById(
+    getLocalStorage("currentProjectIdFromSidebar")
   );
+  if (!wantedProject) {
+    proName.textContent = "NoProjectsYet";
+    threeDotsIcon.classList.add("hide");
+  } else {
+    proName.textContent = wantedProject.projectName;
+    threeDotsIcon.classList.remove("hide");
+  }
+};
+const actionMenuContainer = document.querySelector(".actions-container");
+export const deleteProjectUiAction = (deletedId) => {
+  let projects = getLocalStorage("projects");
+  const project = projects.find((item) => item.ProjectId == deletedId);
+  let currantIndex = projects.indexOf(project);
```

Comment:
Using `.findIndex` combines both operations into one step, making the code more efficient:
```js
const currentIndex = projects.findIndex(item => item.ProjectId === deletedId);

```